RocketChat/Rocket.Chat
# fix: Added check if count is truthy on `cleanRoomHistory`

## Proposed changes (including videos or screenshots)


This PR adds a check when the room is updated inside the function `removeByIdPinnedTimestampLimitAndUsers` which is called on `cleanRoomHistory`; It has been added because even when the room wasn't being updated, the `_updatedAt` was changed, and caused move chats to the top of the list even when nothing has changed there.



## Issue(s)


#27956

## Steps to test or reproduce

- Enable `Retention Policy`.
- For clarity, set "Use Advanced Retention Policy Cron" to every 5 minutes
`*/5 * * * *`
- After 5 minutes, query the database to verify that the `_updatedAt` value has changed in each room:
`db.rocketchat_room.find({ }, {"_updatedAt": 1 }).pretty()`

## Further comments

[TC-488]

[TC-488]: https://rocketchat.atlassian.net/browse/TC-488?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
